src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonReaderTests.cs

```diff
@@ -3765,6 +3781,134 @@ public static void TestBOMWithSingleJsonValue(byte[] utf8BomAndValue, bool isFin
+                {
+                    Assert.False(reader.ValueIsEscaped);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(@"\""")]
+        [InlineData(@"\n")]
+        [InlineData(@"\r")]
+        [InlineData(@"\\")]
+        [InlineData(@"\/")]
+        [InlineData(@"\t")]
+        [InlineData(@"\b")]
+        [InlineData(@"\f")]
+        [InlineData(@"\u6F22\u5B57")]
+        public static void ValueIsEscaped_IsTrueOnEscapedStrings(string jsonString)
+        {
+            string json = $@"""{jsonString}""";
+            JsonTestHelper.AssertWithSingleAndMultiSegmentReader(json, Test);
+
+            static void Test(ref Utf8JsonReader reader)
+            {
+                Assert.True(reader.Read());
+                Assert.Equal(JsonTokenType.String, reader.TokenType);
+                Assert.True(reader.ValueIsEscaped);
+            }
+        }
+
+        [Theory]
+        [InlineData(@"\""")]
+        [InlineData(@"\n")]
+        [InlineData(@"\r")]
+        [InlineData(@"\\")]
+        [InlineData(@"\/")]
+        [InlineData(@"\t")]
+        [InlineData(@"\b")]
+        [InlineData(@"\f")]
+        [InlineData(@"\u6F22\u5B57")]
+        public static void ValueIsEscaped_IsTrueOnEscapedPropertyNames(string jsonString)
+        {
+            string json = $@"{{ ""{jsonString}"" : false }}";
+            JsonTestHelper.AssertWithSingleAndMultiSegmentReader(json, Test);
+
+            static void Test(ref Utf8JsonReader reader)
+            {
+                Assert.True(reader.Read());
+                Assert.True(reader.Read());
+                Assert.Equal(JsonTokenType.PropertyName, reader.TokenType);
+                Assert.True(reader.ValueIsEscaped);
+            }
+        }
+
+        [Theory]
+        [InlineData("null")]
+        [InlineData("false")]
+        [InlineData("true")]
+        [InlineData("42")]
+        [InlineData("\"stringWithoutEscaping\"")]
+        [InlineData("{}")]
+        [InlineData("[]")]
+        [InlineData("/* comment */ null", JsonCommentHandling.Allow)]
+        public static void ValueIsEscaped_AfterEscapedStrings_IsFalseOnNonEscapedTokens(string jsonValue, JsonCommentHandling commentHandling = default)
+        {
+            string json = $@"[""hello\n"", {jsonValue}]";
+            var options = new JsonReaderOptions { CommentHandling = commentHandling };
+            JsonTestHelper.AssertWithSingleAndMultiSegmentReader(json, Test, options);
+
+            static void Test(ref Utf8JsonReader reader)
+            {
+                Assert.True(reader.Read());
+                Assert.True(reader.Read());
+                Assert.Equal(JsonTokenType.String, reader.TokenType);
+                Assert.True(reader.ValueIsEscaped);
+
+                while (reader.Read())
+                {
+                    Assert.False(reader.ValueIsEscaped);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("null")]
```

Comment:
I think those cases repeat in couple of places, it would be better to factor them out into [MemberData(...)]

Reply:
I don't think there is value in extracting to MemberData for only two duplications. In any case it's testing different methods with slightly different semantics, so the fact that there is complete overlap is pure coincidence.